projects/sandbox/train/train/data_structures.py

```diff
@@ -202,3 +213,29 @@ def forward(self, X, y):
             mask = torch.rand(size=X.shape[:-1]) < self.prob
             X[mask] = X[mask].flip(-1)
         return X, y
+
+
+class ChannelSwapper(torch.nn.Module):
+    """
+    Data augmentation module that randomly swaps channels
+    of a fraction of batch elements.
+
+    Args:
+        frac:
+            Fraction of batch that will have channels swapped.
+    """
+
+    def __init__(self, frac: float = 0.5):
+        super().__init__()
+        self.frac = frac
+
+    def forward(self, X):
+        num = int(X.shape[0] * self.frac)
+        num = num if not num % 2 else num + 1
+        indices = torch.randperm(X.shape[0])[:num]
+        channel = torch.randint(X.shape[1], size=(num // 2,))
+        print(X.shape, channel.shape, num)
+        X[indices[-num // 2 :], channel] = X[indices[: num // 2], channel]
+        X[indices[: num // 2], channel] = X[indices[-num // 2 :], channel]
```

Comment:
This won't change anything, because of the previous line. We'll need to make a copy of `X` first.

Reply:
Good call

Reply:
Arrived on an implementation that removes the need to copy

Reply:
Nice, I like it. Solves the circularity of the testing, too.